collector/generic_collector_test.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"io/ioutil"
+
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+)
+
+func TestConfigWithErrors(t *testing.T) {
+	assert := assert.New(t)
+
+	invalidJSON := "{"
```

Comment:
You should be able to use a raw string for this:

http://play.golang.org/p/9xLuph6MEQ

Makes it more readable :)